main.c

```diff
@@ -122,17 +122,51 @@
 uint8_t value_out = 0;
 #endif
 
-#if MICROPY_ENABLE_PYSTACK
-static size_t PLACE_IN_DTCM_BSS(_pystack[CIRCUITPY_PYSTACK_SIZE / sizeof(size_t)]);
+#if MICROPY_ENABLE_PYSTACK && CIRCUITPY_OS_GETENV
+#include "shared-module/os/__init__.h"
 #endif
 
+typedef struct {
+    #if MICROPY_ENABLE_PYSTACK
+    supervisor_allocation *pystack;
+    #endif
+    supervisor_allocation *heap;
+} stacks;
+
 static void reset_devices(void) {
     #if CIRCUITPY_BLEIO_HCI
     bleio_reset();
     #endif
 }
 
-STATIC void start_mp(supervisor_allocation *heap) {
+#if MICROPY_ENABLE_PYSTACK
```

Comment:
I don't think you meant to have the #if around this function.
You should build this change without MICROPY_ENABLE_PYSTACK to ensure it still builds.

Reply:
Fixed the `#if`'s, I will build stackless in a bit.

Reply:
Stackless works just fine. No memory is allocated for pystack.